### ISSUE TITLE
Blank tiles (tile has no (valid) foreground nor background)

### ISSUE THREAD

**Issue by scampada** [CleverRaven/Cataclysm-DDA#12873]
I'm using cdda-0.C.
The game screen is black, if I change output to tiles=false it's drawed with text and works properly.
Anything else doesn't work, no matter which tileset I'm trying or whether software rendering is set or not.
If I press keys when seeing black screen, it may draw strange blue squares and that's all. I can also invoke an inventory screen, but when I close it, it doesn't disappear properly but its parts remain erratically on the display.
The main menu and all menu stuff are displayed properly.
* Tiny Core Linux
* Xvesa server
* No window manager
* SDL; SDL2; SDL2_mixer built manually OK; SDL2_ttf built manually OK; SDL2_image built manually OK; lua-lib is not liblua5.2.so.0 but liblua.so.5.2, created a softlink liblua5.2.so.0->liblua.so.5.2; all other stuff.

What is important: I'm also using dosbox which uses SDL too, and Angband\FAangbband both with tilesets turned on work fine. I think they are also using SDL.

Here are some strings from my debug.log:

13:34:3.504 SDL :
blah blah numbers numbers
>>
>>
src/cata_tiles.cpp:338: tile tr_boobytrap has no (valid) foreground nor background

And there is a huge amount of these strings, telling that the whole throng of tiles have no foreground nor background.

Also, if I press movement keys repeatedly, there may appear some strange blue tiles on my display, moreover, if I press left, there appears a new blue square to the right, and if I press up it appears  under the previous one. Here's a screenshot. Does it mean that at least SDL drawing functions work properly but tiles do not load?
https://www.dropbox.com/s/b5lqb1xb72bsq8q/cdda-bug3.png?dl=0

**Comment by scampada** [comment]
If I open the help in tiles mode and then close it there remains a part of help screen in the tiles area:
https://www.dropbox.com/s/ce22vqcrcurouge/cdda-bug4.png?dl=0

**Comment by BevapDin** [comment]
Just a guess (it's not easy to diagnose this without the full debug log): is there anything about png in the log? It looks like the game did not load the image file and therefor has no tiles to display (which also makes all tile ids invalid).

**Comment by scampada** [comment]
I'll take a look. And I will upload my debug.log and provide a link here a bit later.

**Comment by kevingranade** [comment]
Two things that come to mind are permissions and inability to decode the
png files, though the only instance of the latter I've seen presented as a
crash.

**Comment by scampada** [comment]
Echo-1 to Lair, reporting.
https://www.dropbox.com/s/cl8rmy4bfmtp7kr/debug.log?dl=0

My log file. No sign of any message concerning png. I've checked permissions hours ago! At least they coincide with my u:g. I'm tc:staff, they're all tc:staff. All have rwrr (I mean tilesets and json files).

Please turn your attention to the message of been unable to load wav file. There is a path that is looking somewhat like /data//sound/Dark_Days_Ahead_demo_2.wav. I looked after this file, it doesn't exist. But those two slashes. Aren't they suspicious?

**Comment by scampada** [comment]
I think it's not a trivial problem with loading the file but something more complex. There is something strange happening when I press movement keys in the blank screen. After a while blue squares begin appearing in the opposite direction to key pressed. I press left - a square appears to the right of previous. I press up - it appears under that. But not necessary! Just now I pressed right many times, so squares drew a line from right to left and then... they didn't stop but started again... from the upper "string", again from right to left. Then they drew a half-line stopped in the middle. It's really weird, not only tiles not loading.
Though, if I set tiles off, I can play in ASCII, but... Those tiles are so cool...
